internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java

```diff
@@ -215,6 +218,171 @@ protected void runTest(
     }
   }
 
+  private void runTestWithClientException(
+      ClientConfig.ClientConfigBuilder clientConfigBuilder,
+      boolean batchGet,
+      int batchGetKeySize,
+      Consumer<MetricsRepository> statsValidation,
+      Optional<AvroGenericStoreClient> vsonThinClient) throws Exception {
+    MetricsRepository metricsRepositoryForGenericClient = new MetricsRepository();
+    AvroGenericStoreClient<String, GenericRecord> genericFastClient =
+        getGenericFastClient(clientConfigBuilder, metricsRepositoryForGenericClient, true);
+
+    AvroGenericStoreClient<String, Object> genericFastVsonClient = null;
+    // Construct a Vson store client
+    genericFastVsonClient =
+        getGenericFastVsonClient(clientConfigBuilder.clone(), new MetricsRepository(), vsonThinClient, true);
+    try {
+      if (batchGet) {
+        // test batch get of size 2 (current default max)
+        if (batchGetKeySize == 2) {
+          for (int i = 0; i < recordCnt - 1; ++i) {
+            String key1 = keyPrefix + i;
+            String key2 = keyPrefix + (i + 1);
+            Set<String> keys = new HashSet<>();
+            keys.add(key1);
+            keys.add(key2);
+            try {
+              genericFastClient.batchGet(keys).get();
+              fail();
+            } catch (VeniceClientException e) {
+              assertTrue(e.getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+            } catch (ExecutionException e) {
+              assertTrue(e.getCause().getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+            }
+
+            // Test Vson client
+            try {
+              genericFastVsonClient.batchGet(keys).get();
+              fail();
+            } catch (VeniceClientException e) {
+              assertTrue(e.getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+            } catch (ExecutionException e) {
+              assertTrue(e.getCause().getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+            }
+          }
+        } else if (batchGetKeySize == recordCnt) {
```

Comment:
I don't think the case where `batchGetKeySize==100` is ever run since we only do the case where `batchGetKeySize==2`. Should I remove this case or leave it in so once we enable for other sizes we have the code?

Reply:
Can we double check with @m-nagarajan? Since we are actually trying to move away from the 2 key batch get (temporary implementation) and use the more generic batch get that actually performs the scatter gather.

Reply:
Talked to Manoj and we decided to enable the `batchGetKeySize==100` case for the tests